pyodide_build/tests/recipe/test_bash_runner.py

```diff
@@ -53,11 +53,17 @@ def test_subprocess_with_shared_env_logging(capfd, tmp_path):
         dir.mkdir()
         p.run("echo 1000", script_name="test script", cwd=dir)
         cap = capfd.readouterr()
-        assert [l.strip() for l in cap.out.splitlines()] == [
-            "Running test script in",
-            str(dir),
-            "1000",
-        ]
+
+        # Clean and join all outputs to handle potential line breaks,
+        # and check for path components, instead of the exact path.
+        assert "Running test script in" in cap.out
+
+        # Normalize whitespace for path comparison
+        normalized_output = "".join(cap.out.split())
```

Comment:
The old code is significantly easier to read. What goes wrong with `str(dir)` on mac? Could you only normalize the output and not what we are comparing it to? Like:
```py
normalized_dir = str(dir)
assert normalize_output(cap.out) == [ "Running test script in", normalized_dir, "1000" ]
```
and them make `normalize_output` do whatever you need to clean it up?